language/libra-vm/src/libra_vm.rs

```diff
@@ -78,57 +73,31 @@ impl LibraVM {
         self.load_configs_impl(&RemoteStorage::new(state))
     }
 
-    fn on_chain_config(&self) -> VMResult<&OnlineConfig> {
+    fn on_chain_config(&self) -> VMResult<&VMConfig> {
         self.on_chain_config
             .as_ref()
             .ok_or_else(|| VMStatus::new(StatusCode::VM_STARTUP_FAILURE))
     }
 
     fn load_configs_impl(&mut self, data_cache: &dyn RemoteCache) {
-        self.gas_schedule = self.fetch_gas_schedule(data_cache).ok();
-        self.on_chain_config = OnlineConfig::load_on_chain_config(data_cache);
-    }
-
-    fn fetch_gas_schedule(&mut self, data_cache: &dyn RemoteCache) -> VMResult<CostTable> {
-        let address = account_config::association_address();
-        let mut ctx = SystemExecutionContext::new(data_cache, GasUnits::new(0));
-        let gas_struct_ty = self
-            .move_vm
-            .resolve_struct_def_by_name(&GAS_SCHEDULE_MODULE, &GAS_SCHEDULE_NAME, &mut ctx, &[])
-            .map_err(|_| {
-                VMStatus::new(StatusCode::GAS_SCHEDULE_ERROR)
-                    .with_sub_status(sub_status::GSE_UNABLE_TO_LOAD_MODULE)
-            })?;
-
-        let access_path = create_access_path(address, gas_struct_ty.into_struct_tag()?);
-
-        let data_blob = data_cache
-            .get(&access_path)
-            .map_err(|_| {
-                VMStatus::new(StatusCode::GAS_SCHEDULE_ERROR)
-                    .with_sub_status(sub_status::GSE_UNABLE_TO_LOAD_RESOURCE)
-            })?
-            .ok_or_else(|| {
-                VMStatus::new(StatusCode::GAS_SCHEDULE_ERROR)
-                    .with_sub_status(sub_status::GSE_UNABLE_TO_LOAD_RESOURCE)
-            })?;
-        let table: CostTable = lcs::from_bytes(&data_blob).map_err(|_| {
-            VMStatus::new(StatusCode::GAS_SCHEDULE_ERROR)
-                .with_sub_status(sub_status::GSE_UNABLE_TO_DESERIALIZE)
-        })?;
-
-        Ok(table)
+        self.on_chain_config = VMConfig::fetch_config(data_cache);
+        self.version = LibraVersion::fetch_config(data_cache);
     }
 
     pub fn get_gas_schedule(&self) -> VMResult<&CostTable> {
-        self.gas_schedule.as_ref().ok_or_else(|| {
-            VMStatus::new(StatusCode::VM_STARTUP_FAILURE)
-                .with_sub_status(sub_status::VSF_GAS_SCHEDULE_NOT_FOUND)
-        })
+        self.on_chain_config
+            .as_ref()
+            .map(|config| &config.gas_schedule)
+            .ok_or_else(|| {
+                VMStatus::new(StatusCode::VM_STARTUP_FAILURE)
+                    .with_sub_status(sub_status::VSF_GAS_SCHEDULE_NOT_FOUND)
+            })
     }
 
     pub fn get_libra_version(&self) -> VMResult<LibraVersion> {
-        Ok(self.on_chain_config()?.version.clone())
+        self.version
+            .clone()
+            .ok_or_else(|| VMStatus::new(StatusCode::VM_STARTUP_FAILURE))
```

Comment:
Nit: Maybe add a sub status here stating that it was unable to get the libra version?